python/python/lance/dataset.py

```diff
@@ -60,6 +60,13 @@
         Iterable[RecordBatch],
         pa.RecordBatchReader,
     ]
+    QueryVectorLike = Union[
+        pd.Series,
+        pa.Array,
+        pa.Scalar,
```

Comment:
Why can we use scalar here?

Reply:
It's to handle FixedSizeListScalar (and fixed shape tensor scalar) Pretty often I find myself doing something like:

```
queries = dataset.take(random_indices)
for query in queries:
  # Here `query` is a FixedSizeListScalar
  ...
```